Notice/Event.swift

```diff
@@ -0,0 +1,32 @@
+public protocol Event {
+    typealias ValueType
+    typealias HandlerType
+
+    var value: ValueType { get }
+    func handle(handler: HandlerType)
+}
+
+public struct NewEvent<T>: Event {
+    public typealias HandlerType = T -> Void
+    public let value: T
+
+    public func handle(handler: HandlerType) {
+        handler(value)
+    }
+}
+
+public struct NewOldEvent<T>: Event {
+    public typealias HandlerType = (old: T?, new: T) -> Void
+
+    public let value: T
+    public let oldValue: T?
+
+    init(value: T, oldValue: T? = nil) {
+        self.value = value
+        self.oldValue = oldValue
+    }
+
+    public func handle(handler: HandlerType) {
```

Comment:
Same deal here, let's try to get rid of the typealiases and let the types in the protocol conformance dictate the types.